### PR TITLE
feat: [automl] config oracle cubic on top of quadratic

### PR DESCRIPTION
**Pull request by lalo**

### CODE REVIEW

File: vowpalwabbit/core/src/reductions/details/automl/automl_oracle.cc [hunk 1]
```diff
@@ -178,64 +223,17 @@ void config_oracle<oracle_impl>::insert_config(set_ns_list_t&& new_elements,
   float priority = (*calc_priority)(configs[valid_config_size], ns_counter);
   index_queue.push(std::make_pair(priority, valid_config_size));
   ++valid_config_size;
+  return true;
 }
 
-// This will generate configs based on the current champ. These configs will be
```
Comment (bassmang) [inline vowpalwabbit/core/src/reductions/details/automl/automl_oracle.cc:183]: Can we keep/update this comment where needed?

**Review by bassmang — APPROVED** [review]

File: test/unit_test/automl_test.cc [hunk 2]
```diff
@@ -605,6 +610,172 @@ BOOST_AUTO_TEST_CASE(one_diff_impl_unittest_w_iterations)
       test_hooks, num_iterations, seed);
 }
 
+BOOST_AUTO_TEST_CASE(qbase_unittest_w_iterations)
```
Comment (jackgerrits) [inline test/unit_test/automl_test.cc:613]: New tests should be added to `vowpalwabbit/core/tests`

**Review by jackgerrits — CHANGES_REQUESTED** [review]

File: vowpalwabbit/core/src/reductions/details/automl/automl_oracle.cc (hunk 1 above)
Comment (lalo) [reply vowpalwabbit/core/src/reductions/details/automl/automl_oracle.cc:183]: done

File: test/unit_test/automl_test.cc (hunk 2 above)
Comment (lalo) [reply test/unit_test/automl_test.cc:619]: I'll migrate all the file, since this is almost an exact copy of another test on this same file.

**Pull request merged**